### ISSUE TITLE
"Experiments setup" and "Metrics" sections

### ISSUE THREAD

**Issue by AAAAA-pn** [xiaochengfuhuo/TropiCycloneNet#3]
Hello, author,

First, thank you for making your code open-source.

Regarding the paper "Benchmark dataset and deep learning method for global tropical cyclone forecasting," I was wondering if you could provide the corresponding model testing code for the "Experiments setup" and "Metrics" sections?

**Comment by Nanming666** [comment]
I have also encountered the same problem, and the calculated result is slightly different from the one mentioned in the article. It is possible that my calculation method is incorrect. I also hope that the author can provide the test code for our reference

**Comment by xiaochengfuhuo** [comment]
Hello,

Thank you for your kind words and for your interest in our work.

The test code is still being organized. For now, you can refer to  https://github.com/Zjut-MultimediaPlus/MGTCF/blob/main/scripts/evaluate_model_Me.py — the process is the same for both.

**Comment by xiaochengfuhuo** [comment]
This code maybe more suitable  https://github.com/xiaochengfuhuo/TropiCycloneNet/blob/main/scripts/evaluate_model_Me.py

**Comment by AAAAA-pn** [comment]
Thank you for your reply. I have now reproduced your code, and the parameters I used are as follows:

areas = ['EP','NA','NI','SI','SP','WP']
DEFAULT_EPOCHS="102"
DEFAULT_BATCH_SIZE="96"
DEFAULT_G_LR="1e-4"
DEFAULT_D_LR="1e-4"
DEFAULT_MODAL="gph"
These are all the default parameters from the code, but the test results of the trained model are very poor.

checkpoint_with_model_46000.pt
Dataset: /home/data/TCN-data/data1d/bst_divi10_train_val_test_inlcude15_2023, Pred Len: 4
TDR: tensor([ 27.1562,  58.9407,  94.1120, 134.4286])
TDR: tensor([[1.3197, 0.9323],
        [2.6100, 1.8524],
        [4.1912, 2.9693],
        [5.4469, 3.9038]])



The checkpoint_with_model_46000.pt is the best-performing model from my training.

I was wondering if this is because some parameters were selected incorrectly?

**Comment by xiaochengfuhuo** [comment]
This is a known drawback of GANs—training instability. You can adjust the `pi_pre_epoch` parameter (ranging from 1 to 5). If your training result converges too early and the loss fluctuates significantly, it may indicate a training failure. However, if the convergence is relatively slow—for example, reaching a good result around 16,000 epochs—you should be able to achieve decent results.   By default, it shouldn't be trained up to 46,000. Is it 4,600?

**Comment by AAAAA-pn** [comment]
Thank you for your reply. I will try your suggested method of adjusting the pi_pre_epoch parameter (ranging from 1 to 5).

I located the 46,000 checkpoint by searching the training checkpoint directory for the model checkpoint that had the lowest ADE (Average Displacement Error) on the validation set.
